app/src/main/java/com/cp/tasky/auth/shared/data/repository/AuthRepositoryImpl.kt

```diff
@@ -5,20 +5,26 @@ import com.cp.tasky.auth.shared.data.mapper.LoginUserMapper
 import com.cp.tasky.auth.shared.domain.AuthRepository
 import com.cp.tasky.auth.shared.domain.model.LoginResponse
 import com.cp.tasky.auth.shared.domain.model.UserCredential
+import com.cp.tasky.core.util.NetworkResult
 
 class AuthRepositoryImpl(
     private val remoteDataSource: RemoteDataSource
 ) : AuthRepository {
 
-    override suspend fun loginUser(userCredential: UserCredential): LoginResponse {
-        //TODO:  Handle network response state
-        return LoginUserMapper.toLoginResponse(
-            remoteDataSource.loginUser(
+    override suspend fun loginUser(userCredential: UserCredential): NetworkResult<LoginResponse> {
+        return try {
+            val response = remoteDataSource.loginUser(
                 LoginApiBody(
-                    email = userCredential.email,
-                    password = userCredential.password
+                    email = userCredential.email, password = userCredential.password
                 )
             )
-        )
+
+            NetworkResult.Success(
+                LoginUserMapper.toLoginResponse(response)
+            )
+
+        } catch (e: Exception) {
+            NetworkResult.Error(e)
+        }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video I show the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc